purchases/src/test/java/com/revenuecat/purchases/attributes/SubscriberAttributesDeviceCacheTests.kt

```diff
@@ -163,46 +180,249 @@ class SubscriberAttributesDeviceCacheTests {
 
-        assertCapturedEqualsExpected(expectedAttributes)
+        assertCapturedEqualsExpected(mapOf(appUserID to expectedAttributes))
     }
 
     @Test
     fun `clearing caches also clears the subscriber attributes`() {
-        underTest.clearCachesForAppUserID()
-        verify { mockEditor.remove(underTest.subscriberAttributesCacheKey("appUserID")) }
+        val expectedAttributes = mapOf(
+            "tshirtsize" to SubscriberAttribute("tshirtsize", "L", isSynced = true)
+        )
+        mockNotEmptyCache(expectedAttributes)
+        underTest.clearCachesForAppUserID(appUserID)
+        verify {
+            mockEditor.putString(
+                "com.revenuecat.purchases.$apiKey.subscriberAttributes",
+                JSONObject().also {
+                    it.put("attributes", JSONObject())
+                }.toString()
+            )
+        }
+    }
+
+    @Test
+    fun `Given there are some unsynced attributes for multiple users`() {
+        val attributes = mapOf(
+            "tshirtsize" to SubscriberAttribute("tshirtsize", "L"),
+            "age" to SubscriberAttribute("age", "L", isSynced = true)
+        )
+        mockNotEmptyCacheMultipleUsers(mapOf(
+            appUserID to attributes,
+            "user2" to attributes
+        ))
+        val unsyncedSubscriberAttributes = underTest.getUnsyncedSubscriberAttributes()
+        assertThat(unsyncedSubscriberAttributes.size).isEqualTo(2)
+        unsyncedSubscriberAttributes.forEach { (_, subscriberAttributesForUser) ->
+            assertThat(subscriberAttributesForUser.size).isEqualTo(1)
+            assertThat(subscriberAttributesForUser["tshirtsize"]).isEqualTo(attributes["tshirtsize"])
+        }
+    }
+
+    @Test
+    fun `Given there are some unsynced attributes for one of multiple users`() {
+        val expectedAttribute = SubscriberAttribute("tshirtsize", "L")
+        mockNotEmptyCacheMultipleUsers(mapOf(
+            appUserID to  mapOf(
+                "tshirtsize" to expectedAttribute,
+                "age" to SubscriberAttribute("age", "L", isSynced = true)
+            ),
+            "user2" to  mapOf(
+                "tshirtsize" to SubscriberAttribute("tshirtsize", "L", isSynced = true),
+                "age" to SubscriberAttribute("age", "L", isSynced = true)
+            )
+        ))
+        val unsyncedSubscriberAttributes = underTest.getUnsyncedSubscriberAttributes()
+        assertThat(unsyncedSubscriberAttributes.size).isEqualTo(1)
+        unsyncedSubscriberAttributes.forEach { (_, subscriberAttributesForUser) ->
+            assertThat(subscriberAttributesForUser.size).isEqualTo(1)
+            assertThat(subscriberAttributesForUser["tshirtsize"]).isEqualTo(expectedAttribute)
+        }
+    }
+
+    @Test
+    fun `Given there are no unsynced attributes for multiple users, getting them does't return them`() {
```

Comment:
"getUnsyncedSubscriberAttributes returns an empty list if none are found"?

Reply:
I am laughing when reading the original name of the test